src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs

```diff
@@ -75,12 +76,28 @@ public bool IsAccessible(SmoQueryContext context)
             }
             catch (Exception ex)
             {
-                var error = string.Format(CultureInfo.InvariantCulture, "Failed to get IsAccessible. error:{0} inner:{1} stacktrace:{2}",
-                    ex.Message, ex.InnerException != null ? ex.InnerException.Message : "", ex.StackTrace);
-                Logger.Write(TraceEventType.Error, error);
-                ErrorMessage = ex.Message;
-                return false;
+                // IsAccessible is not set of DW Gen3 so exception is expected in this case
```

Comment:
IsAccessible doesn't really make any sense in Azure. AFAICT if the current connection is to master, has_dbaccess('user database name') always returns null regardless of whether 'user database name' refers to a DW or a DB. 

If you are connecting to the user database  directly, the fact that SqlConnection.Open succeeds means you have access to the database, so why are we bothering to check this property?

Reply:
@shueybubbles I guess with dbpool connections "use master" doesn't work, so the connection will be to the user DB.  I'll see if there is a more generic way to handle this exception that takes into account what you said above.  We don't see this exception connecting to Azure DB normally, so either the property is set and doesn't mean anything or we don't go down this code path.  I'll debug and see what's up.